### PR TITLE
feature:S3C-3537 Implement list users

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
This PR will implement OSIS list users using Vault IAM `list-users`.

* The OSIS `list-users` API can be called with following parameters:
  * offset: The start index of tenants to return (optional)
  * limit: The maximum number of tenants to return (optional)
* IAM `list-users` api will be called using assumed role credentials.
  * If `offset` is present in the request, return the IAM `list-users` result by providing the `offset` value as the `marker` parameter.

### CODE REVIEW

File: osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java [hunk 1]
```diff
@@ -326,6 +343,24 @@ private static OsisTenant toOsisTenant(AccountData account) {
...
+    /**
+     * Converts IAM User object to OSIS User object
+     *
+     * @return the osis User
+     */
+    public static OsisUser toOsisUser(User user, String tenantId) {
+        return new OsisUser()
+                .cdUserId(user.getUserName())
+                .canonicalUserId(user.getUserName())
+                .userId(user.getUserName())
```
Comment (anurag4DSB) [inline osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:355]: Should these be `user.getCanonicalId` and `user.getUserId` respectively?

Comment (saitharunthodupunuru) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:355]: * For OSIS we are assigning `canonicalUserId` with `cd_user_id` i.e., Vault `username` so we can implement `getUser` using `canonicalUserId` on OSIS. (ref: [design for canonical user ID](https://github.com/scality/vmware-ose-scality/blob/documentation/S3C-4180-ose-design-2/Design.md#get-user-w-canonical-id)
* Regarding `userId`, vault `username` is mapped to OSIS `userID`. (Ref: [design for `get-user`](https://github.com/scality/vmware-ose-scality/blob/documentation/S3C-4180-ose-design-2/Design.md#get-user-w-canonical-id)

File: osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java [hunk 2]
```diff
@@ -92,6 +95,20 @@ public static ListAccountsRequestDTO toScalityListAccountsRequest(long limit) {
                 .build();
     }
 
+    /**
+     * Created Vault List Accounts request for List Tenants
```
Comment (anurag4DSB) [inline osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:99]: Should this be ListUsers request?

File: osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java [hunk 3]
```diff
@@ -123,6 +123,7 @@ public PageOfTenants queryTenants(long offset, long limit, String filter) {
                 return pageOfTenants;
 
             } catch (VaultServiceException e) {
+                logger.error("Query Tenants error. Return empty list. Error details: ", e);
```
Comment (anurag4DSB) [inline osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java:126]: If no tenant has been created, do we still log the error as the list would be empty?

File: osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java (hunk 1 above)
Comment (anurag4DSB) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java:355]: I understand now.
Thanks.

**Review by anurag4DSB — APPROVED** [review]

File: osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java (hunk 3 above)
Comment (anurag4DSB) [reply osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java:126]: nvm, I understood.

File: osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java (hunk 2 above)
Comment (saitharunthodupunuru) [reply osis-core/src/main/java/com/scality/osis/utils/ScalityModelConverter.java]: fixed in 5385bbc9255f242f654930813e49a22528eae0d9

File: osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java [hunk 4]
```diff
@@ -323,7 +325,39 @@ public PageOfS3Credentials listS3Credentials(String tenantId, String userId, Lon
...
+            logger.debug("[Vault] List Users response:{}", new Gson().toJson(listUsersResult));
+
+            PageOfUsers pageOfUsers = ScalityModelConverter.toPageOfUsers(listUsersResult, offset, limit, tenantId);
+            logger.info("List Users response:{}", new Gson().toJson(pageOfUsers));
+
+            return  pageOfUsers;
+        } catch (Exception e){
+
+            logger.error("ListUsers error. Returning empty list. Error details: ", e);
+            // For errors, List Tenants should return empty PageOfTenants
```
Comment (ilkescality) [inline osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java:349]:
```suggestion
            // For errors, List Users should return empty PageOfUsers
```

**Review by ilkescality — APPROVED** [review]

**Pull request merged**